PlagiarismPlugin.inc.php

```diff
@@ -64,17 +64,23 @@ function getEnabled($contextId = null) {
 	public function callback($hookName, $args) {
 		$request = Application::getRequest();
 		$context = $request->getContext();
+		$contextPath = $context->getPath();
 		$submissionDao = Application::getSubmissionDAO();
 		$submission = $submissionDao->getById($request->getUserVar('submissionId'));
 		$publication = $submission->getCurrentPublication();
 
 		require_once(dirname(__FILE__) . '/vendor/autoload.php');
 
-		$ithenticate = new \bsobbe\ithenticate\Ithenticate(
-			Config::getVar('ithenticate', 'username'),
-			Config::getVar('ithenticate', 'password')
-		);
-
+		// try to get credentials for current context otherwise use default config
+		$username = Config::getVar('ithenticate', 'username['.$contextPath.']');
+		if (!$username) {
```

Comment:
It's best to avoid loose comparisons of values; it's unlikely, but if someone has a username or password of `0`, then this kind of check will not have the expected effect. I'd suggest using the `$default` parameter to `Config::getVar` to get a fallback for the username and password configuration:
```php
$username = Config::getVar(
    'ithenticate', "username[$contextPath]",
    Config::getVar('ithenticate', 'username')
);
```